compiler_opt/rl/corpus.py

```diff
@@ -14,12 +14,105 @@
 # limitations under the License.
 """ModuleSpec definition and utility command line parsing functions."""
 
+from absl import logging
 from dataclasses import dataclass
+from typing import Iterable, Optional, Tuple, List
+
+import os
+import tensorflow as tf
 
 
 @dataclass(frozen=True)
 class ModuleSpec:
   """Dataclass describing an input module and its compilation command options.
   """
   name: str
-  has_thinlto: bool = False
+  _exec_cmd: Tuple[str, ...] = ()
+
+  def cmd(self) -> List[str]:
+    """Retrieves the compiler execution options.
+    """
+    # Using a getter as this will be modified in the future
+    return list(self._exec_cmd)
+
+
+def build_modulespecs_from_datapath(
+    data_path: str,
+    additional_flags: Tuple[str, ...] = (),
+    delete_flags: Tuple[str, ...] = ()
+) -> List[ModuleSpec]:
+  module_paths: List[str] = _load_module_paths(data_path)
+
+  has_thinlto: bool = _has_thinlto_index(module_paths)
+
+  module_specs: List[ModuleSpec] = []
+
+  # This takes ~7s for 30k modules
```

Comment:
could be parallelized?

Reply:
(multi-threading/processing) overhead meant it only became ~4s with diminishing returns with more workers, didn't seem worth it